cmd/gateway-azure.go

```diff
@@ -426,27 +451,58 @@ func (a *azureObjects) PutObjectPart(bucket, object, uploadID string, partID int
 		return info, traceError(InvalidUploadID{})
 	}
 	var sha256Writer hash.Hash
+	var md5sumWriter hash.Hash
+	var etag string
+
+	var writers []io.Writer
+
 	if sha256sum != "" {
 		sha256Writer = sha256.New()
+		writers = append(writers, sha256Writer)
 	}
 
-	teeReader := io.TeeReader(data, sha256Writer)
+	if md5Hex != "" {
+		md5sumWriter = md5.New()
+		writers = append(writers, md5sumWriter)
+		etag = md5Hex
+	} else {
+		// Generate random ETag.
+		u, err := uuid.New()
```

Comment:
just use mustGetUUID() or just have a variant with `getUUID()` to return error if needed.